### PR TITLE
Api 13395 add rating information service

### PR DESCRIPTION
**Pull request by KevinCranmer**
This PR is to add the readAncillaryDecisions method from the RatingInformationService. This is for the upcoming VA Letter Generator API.

I have tested this new method with the [bgs-client](https://github.com/department-of-veterans-affairs/lighthouse-benefits-gateway-services/pull/23) through our new api and confirmed it's working.

I am unsure about versioning but believe I have followed previous PRs examples correctly.

### CODE REVIEW

File: mock-bgs-tests/pom.xml [hunk 1]
```diff
@@ -39,6 +34,12 @@
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>gov.va.api.lighthouse</groupId>
+      <artifactId>bgs-model</artifactId>
+      <version>${bgs-model.version}</version>
+      <scope>test</scope>
```
Comment (esubocz-va) [inline mock-bgs-tests/pom.xml:41]: why has a test scope been added?

File: mock-bgs-tests/pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
   <version>3.0.11-SNAPSHOT</version>
   <packaging>jar</packaging>
   <properties>
-    <bgs-model.version>3.0.1</bgs-model.version>
+    <bgs-model.version>5.0.2-SNAPSHOT</bgs-model.version>
```
Comment (esubocz-va) [inline mock-bgs-tests/pom.xml:15]: will need to be updated from snapshot version

File: mock-bgs/pom.xml [hunk 3]
```diff
@@ -13,7 +13,7 @@
   <packaging>jar</packaging>
   <properties>
     <apache.ws.security>1.6.19</apache.ws.security>
-    <bgs-model.version>4.0.0</bgs-model.version>
+    <bgs-model.version>5.0.2-SNAPSHOT</bgs-model.version>
```
Comment (esubocz-va) [inline mock-bgs/pom.xml:16]: snapshot version

File: mock-bgs/src/main/resources/read-ancillary-decisions-data/600055365.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (esubocz-va) [inline mock-bgs/src/main/resources/read-ancillary-decisions-data/600055365.xml:1]: by only adding one xml file, you are only mocking a single user, 600055365. seems like overkill to be creating a whole H2 table for a single user. something like [this](https://github.com/department-of-veterans-affairs/lighthouse-mock-benefits-gateway-services/blob/master/mock-bgs/src/main/java/gov/va/api/lighthouse/mockbgs/MedicalTreatmentFacilityLookupEndpoint.java) may be a better approach. 

however, there is also benefit in having different data in support of multiple use cases. so if you have other use cases you'd like to cover, adding additional test data would also be a good approach.

**Comment by dbrown-va** [comment]
This PR has a couple conflicts following the last merge that will also need to be addressed.

File: mock-bgs/src/main/resources/read-ancillary-decisions-data/600055365.xml (hunk 4 above)
Comment (aaronhietpas) [reply mock-bgs/src/main/resources/read-ancillary-decisions-data/600055365.xml:1]: added more examples with different ancilyRatingDecn fields, including one that does not have a PT35 type, that are used to determine Ch35 eligibility.

**Comment by aaronhietpas** [comment]
> This PR has a couple conflicts following the last merge that will also need to be addressed.

Merged master into this branch and fixed conflicts

File: mock-bgs/src/main/resources/read-ancillary-decisions-data/153759462.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (esubocz-va) [inline mock-bgs/src/main/resources/read-ancillary-decisions-data/153759462.xml:1]: would rather `read-ancillary-decisions` be a sub directory of data along with the others.

**Review by esubocz-va — APPROVED** [review]

**Pull request merged**